packages/waku/src/router/create-pages.ts

```diff
@@ -457,3 +471,357 @@ export function createPages<
+        throw new Error(`Duplicated dynamic path: ${layout.path}`);
+      }
+      const pathSpec = parsePathWithSlug(layout.path);
+      dynamicLayoutPathMap.set(layout.path, [pathSpec, layout.component]);
+    } else {
+      throw new Error('Invalid layout configuration');
+    }
+  };
+
+  const createRoot: CreateRoot = (root) => {
+    if (configured) {
+      throw new Error('createRoot no longer available');
+    }
+    if (rootItem) {
+      throw new Error(`Duplicated root component`);
+    }
+    if (root.render === 'static' || root.render === 'dynamic') {
+      rootItem = root;
+    } else {
+      throw new Error('Invalid root configuration');
+    }
+  };
+
+  let ready: Promise<AllPages | void> | undefined;
+  const configure = async () => {
+    if (!configured && !ready) {
+      ready = fn({ createPage, createLayout, createRoot });
+      await ready;
+      configured = true;
+    }
+    await ready;
+  };
+
+  const getLayouts = (path: string): string[] => {
+    const pathSegments = path.split('/').reduce<string[]>(
+      (acc, segment, index) => {
+        if (segment === '') {
+          return acc;
+        }
+        acc.push(acc[index - 1] + '/' + segment);
+        return acc;
+      },
+      ['/'],
+    );
+
+    return pathSegments.filter(
+      (segment) =>
+        dynamicLayoutPathMap.has(segment) ||
+        staticComponentMap.has(joinPath(segment, 'layout').slice(1)), // feels like a hack
+    );
+  };
+
+  const definedRouter = new_defineRouter({
+    getPathConfig: async () => {
+      await configure();
+      const paths: {
+        pattern: string;
+        path: PathSpec;
+        components: Record<string, { isStatic: boolean }>;
+        noSsr: boolean;
+      }[] = [];
+      const rootIsStatic = !rootItem || rootItem.render === 'static';
+
+      for (const [path, pathSpec] of fixedPathSet) {
+        const noSsr = noSsrSet.has(pathSpec);
+        const isStatic = (() => {
+          for (const [_, [layoutPathSpec]] of dynamicLayoutPathMap) {
+            if (hasPathSpecPrefix(layoutPathSpec, pathSpec)) {
+              return false;
+            }
+          }
+          return true;
+        })();
+
+        const pattern = path2regexp(parsePathWithSlug(path));
+
+        const layoutPaths = getLayouts(pattern);
+
+        paths.push({
+          pattern,
+          path: pathSpec,
+          components: {
+            ...layoutPaths.reduce<Record<string, { isStatic: boolean }>>(
+              (acc, lPath) => {
+                acc[`layout:${lPath}`] = {
+                  isStatic: !dynamicLayoutPathMap.has(lPath),
+                };
+                return acc;
+              },
+              {},
+            ),
+            root: { isStatic: rootIsStatic },
+            [`route:${path}`]: { isStatic },
+            [`page:${path}`]: { isStatic },
+          },
+          noSsr,
+        });
+      }
+      for (const [path, [pathSpec]] of dynamicPagePathMap) {
```

Comment:
Oh, I see. I think I misunderstood the code strucuture.

Reply:
yeah.. their split by map so isStatic is assumed inside the block... seems fine to me